File: kisaansaathi/lib/models/worker.dart
class Worker {
  final String id;
  final String name;
  final String phoneNumber;
  final String language;
  final String? profileImageUrl;
  final DateTime createdAt;

  Worker({
    required this.id,
    required this.name,
    required this.phoneNumber,
    required this.language,
    this.profileImageUrl,
    required this.createdAt,
  });

  factory Worker.fromJson(Map<String, dynamic> json) {
    return Worker(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      phoneNumber: json['phoneNumber'] ?? '',
      language: json['language'] ?? 'English',
      profileImageUrl: json['profileImage']?['url'],
      createdAt: DateTime.parse(
        json['createdAt'] ?? DateTime.now().toIso8601String(),
      ),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'name': name,
      'phoneNumber': phoneNumber,
      'language': language,
      'profileImage': profileImageUrl != null ? {'url': profileImageUrl} : null,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}

File: kisaansaathi/backend/models/Worker.js
const mongoose = require('mongoose');

const workerSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true,
  },
  phoneNumber: {
    type: String,
    required: true,
    unique: true,
    trim: true,
    validate: {
      validator: function(v) {
        return /^\d{10}$/.test(v);
      },
      message: props => `${props.value} is not a valid phone number! Must be 10 digits.`
    }
  },
  language: {
    type: String,
    default: 'English',
  },
  profileImage: {
    url: String,
    publicId: String,
  },
}, {
  timestamps: true,
});

// Index for faster phone number lookups
workerSchema.index({ phoneNumber: 1 });

module.exports = mongoose.model('Worker', workerSchema);
